tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchReindexingTests.java

```diff
@@ -178,8 +179,112 @@ public void testRecreateIndex() throws Exception
 
         // TIDY
         // Restart ElasticsearchConnector.
+        cleanUpIndex();
+        AlfrescoStackInitializer.liveIndexer.start();
+
+    }
+
+    /**
+     * Common testing method for reindexing enabled and disabled features tests.
+     * @param metadataIndexingEnabled Reindexing metadata is enabled when true, disabled when false
+     * @param contentIndexingEnabled Reindexing content is enabled when true, disabled when false
+     * @param pathIndexingEnabled Reindexing path is enabled when true, disabled when false
+     * @param queryString Verification query string. It may include a <DOCUMENT_NAME> mark that is replaced by the actual document name created.
+     * @param expectingDocNameAsResult Result from verification query string is the name of the document created when true, empty result when false.
+     */
+    private void internalTestEnabledFeatures(
+        Boolean metadataIndexingEnabled,
+        Boolean contentIndexingEnabled,
+        Boolean pathIndexingEnabled,
+        String queryString,
+        Boolean expectingDocNameAsResult
+    ) throws Exception
+    {
+
+        // Initial timestamp for reindexing by date: this will save reindexing time for these tests
```

Comment:
Does it actually save a measurable reindexing time?  If it does then there are probably other tests that could benefit from this, if not then IMO we should simplify by removing it (since we can reindex everything by just supplying `reindexByDate`).

Reply:
It's saving 2-3 seconds per test... Is that worthy?

Reply:
That sounds like a really good saving!  We should (perhaps as a separate PR) make the same change to the other tests if we can.